pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -410,13 +420,11 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 	c.InstanceTags = instance.Tags
 	c.MetricTags = instance.MetricTags
 
-	c.OidConfig.addScalarOids(parseScalarOids(c.Metrics, c.MetricTags))
-	c.OidConfig.addColumnOids(parseColumnOids(c.Metrics))
+	c.addUptimeMetric()
```

Comment:
note: moved before:

```
	if c.CollectDeviceMetadata {
		c.OidConfig.addScalarOids(metadata.ScalarOIDs)
		c.OidConfig.addColumnOids(metadata.ColumnOIDs)
	}
```

to avoid having to call `c.OidConfig.addScalarOids` in `addUptimeMetric()` method.

Reply:
Yes, why not :) now that addUptimeMetric only contain one line.

Reply:
This `NewCheckConfig` method is already very long (>200 lines). I would rather start splitting it into multiple smaller ones... My two cents.

Reply:
Agree, `NewCheckConfig` is becoming a bit long, can/should be refactored as a separate PR.